STM/src/test/java/org/jboss/stm/TaxonomyNestedTest.java

```diff
@@ -0,0 +1,407 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2018, JBoss Inc., and individual contributors as indicated
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.stm;
+
+import com.arjuna.ats.arjuna.AtomicAction;
+import com.arjuna.ats.arjuna.coordinator.ActionStatus;
+import junit.framework.TestCase;
+import org.jboss.stm.annotations.Nested;
+import org.jboss.stm.annotations.ReadLock;
+import org.jboss.stm.annotations.Transactional;
+import org.jboss.stm.annotations.WriteLock;
+import org.jboss.stm.internal.RecoverableContainer;
+import org.junit.BeforeClass;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+
+// these tests are discussed in the article https://jbossts.blogspot.com/2018/09/tips-on-how-to-evaluate-stm.html
```

Comment:
* It coudl be a good to add a summary of what the tests are about and then refers to details with the link to the article.
* As this is the comment covering the whole test class it would be good to use the `/** */ format of the comment to that being the generated javadoc.
* It could be better to use the link with `<a href="...`">...</a>` as the javadoc is then generated with the clicable link. 

Maybe I would consider to use the javadoc format of comments (`/** */` instead of `//`) in other places in the file as well. I'm not going to list them in this review as that could be intentional as this is just a test case and not an API expected to be used.

Reply:
okay

Reply:
I added javadoc for the class and for each test.

The tests are meant to supplement the article and should be read in an IDE - ie they are not meant for API usage but rather for the twofold purpose of:
- illumination
- validating our STM behaviour